tests/hypervolume_tests/test_utils.py

```diff
@@ -23,3 +28,55 @@ def test_compute_2d() -> None:
         np.random.shuffle(s)
         v = optuna._hypervolume._compute_2d(s, r)
         assert v == n * n - n * (n - 1) // 2
+
+
+def _compute_hssp_truth_and_approx(test_case: np.ndarray, subset_size: int) -> Tuple[float, float]:
+    r = 1.1 * np.max(test_case, axis=0)
+    truth = 0.0
+    for subset in itertools.permutations(test_case, subset_size):
+        truth = max(truth, optuna._hypervolume.WFG().compute(np.asarray(subset), r))
+    indices = optuna._hypervolume.hssp._solve_hssp(
+        test_case, np.arange(len(test_case)), subset_size, r
+    )
+    approx = optuna._hypervolume.WFG().compute(test_case[indices], r)
+    return truth, approx
+
+
+@pytest.mark.parametrize("dim", [2, 3])
+def test_solve_hssp(dim: int) -> None:
+    random.seed(128)
+
+    for i in range(8):
+        subset_size = int(random.random() * i) + 1
+        test_case = np.asarray([[random.random() for _ in range(dim)] for _ in range(8)])
+        truth, approx = _compute_hssp_truth_and_approx(test_case, subset_size)
+        assert approx / truth > 0.6321  # 1 - 1/e
+
+
+def test_solve_hssp_infinite_loss() -> None:
+    random.seed(128)
+
+    subset_size = int(random.random() * 4) + 1
+    test_case = np.asarray([[random.random() for _ in range(2)] for _ in range(8)])
+    test_case = np.vstack([test_case, [float("inf") for _ in range(2)]])
+    truth, approx = _compute_hssp_truth_and_approx(test_case, subset_size)
+    assert np.isinf(truth)
+    assert np.isinf(approx)
+
+    test_case = np.asarray([[random.random() for _ in range(3)] for _ in range(8)])
+    test_case = np.vstack([test_case, [float("inf") for _ in range(3)]])
+    truth, approx = _compute_hssp_truth_and_approx(test_case, subset_size)
+    assert truth == 0
+    assert np.isnan(approx)
+
+    test_case = np.asarray([[random.random() for _ in range(2)] for _ in range(8)])
+    test_case = np.vstack([test_case, [-float("inf") for _ in range(2)]])
+    truth, approx = _compute_hssp_truth_and_approx(test_case, subset_size)
+    assert np.isinf(truth)
+    assert np.isinf(approx)
+
+    test_case = np.asarray([[random.random() for _ in range(3)] for _ in range(8)])
+    test_case = np.vstack([test_case, [-float("inf") for _ in range(3)]])
+    truth, approx = _compute_hssp_truth_and_approx(test_case, subset_size)
+    assert np.isinf(truth)
+    assert np.isinf(approx)
```

Comment:
Probably just forgot but you can remove the additions in this file now that you've added them in `test_hssp.py`.

Reply:
Oops! Fixed! 
Btw, I saw this TODO comment in utils.py
>def _get_observation_pairs(
    study: Study,
    param_names: List[str],
    constant_liar: bool = False,  # TODO(hvy): Remove default value and fix unit tests.

Reply:
You mean in `optuna/samplers/_tpe/sampler.py` ?
If I remember correctly, I wanted to remove the default argument since it's a parameter of a private function, and it made little good (to developers, and none to users). Might actually not be that relevant now that it has an additional parameter with a default value.

> def _get_observation_pairs(
    study: Study,
    param_names: List[str],
    constant_liar: bool = False,  # TODO(hvy): Remove default value and fix unit tests.
    constraints_enabled: bool = False,

https://github.com/optuna/optuna/blob/master/optuna/samplers/_tpe/sampler.py#L586-L590

You don't happen to be looking for another issue to work on? 🙂